core/gui/src/app/common/service/user/user.service.ts

```diff
@@ -92,4 +95,31 @@ export class UserService {
     }
     return { result: true, message: "Username frontend validation success." };
   }
+
+  getAvatar(googleAvatar: string): string | undefined {
+    const cached = this.cache.get(googleAvatar);
+
+    if (cached && Date.now() > cached.expiry) {
+      this.cache.delete(googleAvatar);
+      return undefined;
+    }
+
+    return cached?.url;
+  }
+
+  setAvatar(googleAvatar: string, avatarUrl: string): void {
+    this.cache.set(googleAvatar, {
+      url: avatarUrl,
+      expiry: Date.now() + this.cacheDuration,
+    });
+  }
+
+  hasAvatar(googleAvatar: string): boolean {
+    const cached = this.cache.get(googleAvatar);
+    return !!cached && Date.now() <= cached.expiry;
+  }
+
+  clearCache(): void {
```

Comment:
You don't need to clear cache after user logout since it's unlikely two unrelated users will use the same device and the icon information is not private.